core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -887,8 +888,18 @@ private void connectDownstream() {
             public void packetReceived(Session session, Packet packet) {
                 if (protocol.getState() == ProtocolState.CONFIGURATION) {
                     if (packet instanceof ClientboundFinishConfigurationPacket) {
-                        // Prevent
-                        GeyserSession.this.ensureInEventLoop(() -> GeyserSession.this.sendDownstreamPacket(new ServerboundFinishConfigurationPacket()));
+                        GeyserSession.this.ensureInEventLoop(() -> {
```

Comment:
Now that this is a bit more of an intensive method, the cleaner solution is probably to make a dedicated translator for this packet, and just "return" here so the finish configuration packet is sent on the session thread and not by MCProtocolLib.

Reply:
Reverted it, and added a separate translator